File: data/cpp/tests/test_numpy_where.cpp
#include <gtest/gtest.h>
#include "../src/function_numpy_where.cpp"

TEST(NumpyExampleTest, test_check_number_positive)
{
    NumpyExample calc;
    std::string result = calc.check_number<int>(5);
    EXPECT_EQ(result, "positive");
}

TEST(NumpyExampleTest, test_check_number_negative)
{
    NumpyExample calc;
    std::string result = calc.check_number<int>(-3);
    EXPECT_EQ(result, "negative");
}

TEST(NumpyExampleTest, test_check_number_zero)
{
    NumpyExample calc;
    std::string result = calc.check_number<int>(0);
    EXPECT_EQ(result, "zero");
}

TEST(NumpyExampleTest, test_check_number_large_positive)
{
    NumpyExample calc;
    std::string result = calc.check_number<int>(100);
    EXPECT_EQ(result, "positive");
}

TEST(NumpyExampleTest, test_check_number_large_negative)
{
    NumpyExample calc;
    std::string result = calc.check_number<int>(-100);
    EXPECT_EQ(result, "negative");
}

int main(int argc, char** argv)
{
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/java/tests/test_numpy_where.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_numpy_where {
    @Test
    void test_check_number_positive() {
        function_numpy_where calc = new function_numpy_where();
        String result = calc.check_number(5);
        assertEquals("positive", result);
    }

    @Test
    void test_check_number_negative() {
        function_numpy_where calc = new function_numpy_where();
        String result = calc.check_number(-3);
        assertEquals("negative", result);
    }

    @Test
    void test_check_number_zero() {
        function_numpy_where calc = new function_numpy_where();
        String result = calc.check_number(0);
        assertEquals("zero", result);
    }

    @Test
    void test_check_number_large_positive() {
        function_numpy_where calc = new function_numpy_where();
        String result = calc.check_number(100);
        assertEquals("positive", result);
    }

    @Test
    void test_check_number_large_negative() {
        function_numpy_where calc = new function_numpy_where();
        String result = calc.check_number(-100);
        assertEquals("negative", result);
    }
}